packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -158,43 +159,59 @@ const ProductDetail = () => {
     )
 
     const handleAddToWishlist = (product, variant, quantity) => {
-        createCustomerProductListItem.mutate(
-            {
-                parameters: {
-                    listId: wishlist.id,
-                    customerId
-                },
-                body: {
-                    // NOTE: APi does not respect quantity, it always adds 1
-                    quantity,
-                    productId: variant?.productId || product?.id,
-                    public: false,
-                    priority: 1,
-                    type: 'product'
-                }
-            },
-            {
-                onSuccess: () => {
-                    toast({
-                        title: formatMessage(TOAST_MESSAGE_ADDED_TO_WISHLIST, {quantity: 1}),
-                        status: 'success',
-                        action: (
-                            // it would be better if we could use <Button as={Link}>
-                            // but unfortunately the Link component is not compatible
-                            // with Chakra Toast, since the ToastManager is rendered via portal
-                            // and the toast doesn't have access to intl provider, which is a
-                            // requirement of the Link component.
-                            <Button variant="link" onClick={() => navigate('/account/wishlist')}>
-                                {formatMessage(TOAST_ACTION_VIEW_WISHLIST)}
-                            </Button>
-                        )
-                    })
+        const isItemInWishlist = wishlist?.customerProductListItems?.find(
+            (i) => i.productId === variant?.productId || i.productId === product?.id
+        )
+
+        if (!isItemInWishlist) {
```

Comment:
There are a couple of solutions possible here. We could change the Add To Wishlist into a Remove from Wishlist button or we could disable the button and change the text to say Added To Wishlist.

I opted to go with the simplest approach for now and just show a different toast as this requires the least amount of change (the other solutions will touch more components like product-view and product-tile)

Reply:
I'm good with this simple approach. 

Btw, can you add a test to verify this change?